gem_city.rb

```diff
@@ -0,0 +1,39 @@
+# This class represents the town of GemCity
+# This is a town riddled with crime but we can find out how happy the town is
+class GemCity
+  attr_reader :population
+  attr_accessor :thieves, :officers
+
+  def initialize
+    @thieves = 5
+    @officers = 1
+    @population = 50
+  end
+
+  def happiness_of_town
+    # happiness is random... people don't know what they want!
+    happiness_vals = (1..@population).collect do
+      rand((100 - successful_crime_rate)..100)
+    end
+    happiness = happiness_vals.reduce(0, :+)
+    happiness / 100
+  end
+
+  def successful_crime_rate
+    if @thieves <= 0 || @officers > thieves
```

Comment:
Whenever these `if` conditions get even a little complicated, I think they should be extracted to a private method. What do you think a good name for this condition might be? `no_crime_can_occur?`, `plenty_of_cops?`, something like that.